tests/test_whoami.rs

```diff
@@ -0,0 +1,38 @@
+use common::util::*;
+
+#[test]
+fn test_normal() {
+    let (_, mut ucmd) = at_and_ucmd!();
+
+    let result = ucmd.run();
+    println!("result.stdout = {}", result.stdout);
+    println!("result.stderr = {}", result.stderr);
+    if result.stderr.contains("failed to get username") {
+        return;
+    }
+
+    assert!(result.success);
+    assert!(!result.stdout.trim().is_empty());
+}
+
+#[test]
+fn test_normal_compare_id() {
+    let (_, mut ucmd) = at_and_ucmd!();
+
+    let result = ucmd.run();
+
+    println!("result.stdout = {}", result.stdout);
+    println!("result.stderr = {}", result.stderr);
+    if result.stderr.contains("failed to get username") {
+        return;
+    }
+    assert!(result.success);
+    let ts = TestScenario::new("id");
+    let id = ts.cmd("id").arg("-un").run();
+    if id.stderr.contains("cannot find name for user ID") {
+        // In the CI, some server are failing to return whoami.
+        // As seems to be a configuration issue, ignoring it
+        return;
```

Comment:
I think it would be better to explicitly test for CI before returning without error.
I would just check for `CI` == "true" within the environment variables.

Reply:
Why not :)
Do we have some usage of the CI variable already?

Reply:
I'm not sure. I don't remember seeing it before...

Reply:
Debugging CI isn't easy :p

the CI variable isn't set. So, I am looking at the USER variable which is "runner" and skipping the test only in this case

I think it is a good compromise, ok @rivy ?

Reply:
👍